tests/GltfTests.cpp

```diff
@@ -0,0 +1,150 @@
+#include "CesiumGltfReader/GltfReader.h"
+#include "testUtils.h"
+
+#include "cesium/omniverse/GltfAccessors.h"
+#include "cesium/omniverse/GltfUtil.h"
+
+#include <CesiumGltf/Material.h>
+#include <CesiumGltf/MeshPrimitive.h>
+#include <CesiumGltf/Model.h>
+#include <doctest/doctest.h>
+#include <sys/types.h>
+
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <gsl/span>
+#include <yaml-cpp/yaml.h>
+
+const std::string ASSET_DIR = "tests/testAssets/gltfs";
+const std::string CONFIG_PATH = "tests/configs/gltfConfig.yaml";
+
+// simplifies casting when comparing some material queries to expected output from config
+bool operator==(pxr::GfVec3f v3, std::vector<float> v) {
+    return v.size() == 3 && v3[0] == v[0] && v3[1] == v[1] && v3[2] == v[2];
+}
+
+TEST_SUITE("gltf utils") {
+    TEST_CASE("IndicesAccessor smoke test") {
+        uint64_t data;
+        std::list<uint64_t> dataContainer = {42, 64, 8675309, 21};
+
+        DOCTEST_VALUE_PARAMETERIZED_DATA(data, dataContainer);
+
+        CHECK(cesium::omniverse::IndicesAccessor(data).size() == data);
+    }
+
+    void checkGltfExpectedResults(const std::filesystem::path& gltfFileName, YAML::Node expectedResults) {
+
+        // --- Load Gltf ---
+        std::ifstream gltfStream(gltfFileName, std::ifstream::binary);
+        std::stringstream gltfBuf;
+        gltfBuf << gltfStream.rdbuf();
```

Comment:
Even though this technically works, it's awkward to read binary data as a stringstream. Something like the answer [here](https://stackoverflow.com/a/15366657) is more idomatic.

Reply:
fixed, thanks for the idiomatic tip